core/domain/state_domain_test.py

```diff
@@ -4443,6 +4444,390 @@ def test_traverse_by_schema_and_get(self):
         self.assertEqual(html, ['<p>testing</p>'])
 
 
+class InteractionInstanceDomainTests(test_utils.GenericTestBase):
+
+    def test_interactions_converted_to_proto_correctly(self):
+        interaction = state_domain.InteractionInstance
+        subtitled_html = state_domain.SubtitledHtml(
+            'default_outcome', '')
+        default_outcome = state_domain.Outcome(
+            'end_state_name', subtitled_html, False, [], None, None)
+
+        continue_customization_args = (
+            interaction.convert_customization_args_dict_to_customization_args(
+                'Continue',
+                {
+                    'buttonText': {
+                        'value': {
+                            'content_id': 'ca_buttonText_1',
+                            'unicode_str': 'Click Me!'
+                        }
+                    }
+                }
+            ))
+
+        continue_interaction = state_domain.InteractionInstance(
+            'Continue',
+            continue_customization_args, [], default_outcome, [], [], None)
+        continue_proto = continue_interaction.to_android_interaction_proto()
+        self.assertEqual(
+            type(continue_proto.continue_instance),
+            state_pb2.ContinueInstanceDto
+        )
+
+        fraction_customization_args = (
+            interaction.convert_customization_args_dict_to_customization_args(
+                'FractionInput',
+                {
+                    'requireSimplestForm': {
+                        'value': False
+                    },
+                    'allowImproperFraction': {
+                        'value': True
+                    },
+                    'allowNonzeroIntegerPart': {
+                        'value': True
+                    },
+                    'customPlaceholder': {
+                        'value': {
+                            'content_id': 'ca_customPlaceholder_2',
+                            'unicode_str': '😍😍😍😍'
+                        }
+                    },
+                }
+            ))
+        fraction_interaction = state_domain.InteractionInstance(
+            'FractionInput',
+            fraction_customization_args, [], default_outcome, [], [], None)
+        fraction_input_proto = (
+            fraction_interaction.to_android_interaction_proto())
+        self.assertEqual(
+            type(fraction_input_proto.fraction_input),
+            state_pb2.FractionInputInstanceDto
+        )
+
+        item_customization_args = (
```

Comment:
Here and below: use the full name, e.g. item_selection_customization_args, multiple_choice_customization_args, etc. Clearer for the reader.

Reply:
Fixed